speech/infinite_streaming_from_file/infinite_audio.go

```diff
@@ -0,0 +1,235 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Before executing this program, the sampleRate parameter needs to be adjusted
+// to match your audio file, 16000 is used in this program as an example
+
+// To run this program, you could use the following command:
+// cat [your_audio_filename].wav | go run infinite_audio.go
+
+/*
+The infinite_streaming_from_file program reads audio data from file
+and pipes it to Google Speech API to output the transcript infinitely.
+*/
+package main
+
+// [START speech_transcribe_streaming]
+import (
+	"context"
+	"io"
+	"log"
+	"math"
+	"os"
+	"sync"
+	"time"
+
+	speech "cloud.google.com/go/speech/apiv1"
+	color "github.com/fatih/color"
+	speechpb "google.golang.org/genproto/googleapis/cloud/speech/v1"
+)
+
+const (
+	sampleRate            = 16000                                      // Audio sample rate hertz.
+	inputChannel          = 1                                          // Number of audio input channel.
+	outputChannel         = 0                                          // Number of audio output channel.
+	bytesPerSample        = 2                                          // Number of bytes each sample consists of.
+	bytesPerSecond        = sampleRate * inputChannel * bytesPerSample // Number of bytes each second audio consists of.
+	streamTimeLimit       = 290 * time.Second                          // Streaming API Limit( ≈ 5 mins).
+	sampleDuration  int64 = 1000                                       // The duration of each sample is 1000 ms.
+)
+
+func main() {
+	// Define data storage buffer.
+	framesPerBuffer := make([]byte, bytesPerSecond)
+	processingBuffer := make([][]byte, 0)
+
+	// Define time variable.
+	streamDeadline := time.Now().Add(streamTimeLimit)
+	processingBufferStart := int64(0)
+	restartCounter := 0
+
+	// Define flag variable.
+	streamIsNull := true
+	exitFlag := false
+	inputIsNull := false
+
+	// Define speech stream variable.
+	var ctx context.Context
+	var client *speech.Client
+	var stream speechpb.Speech_StreamingRecognizeClient
+	var err error
+
+	// There are three functions in the loop:  (Initialize Stream) ->   [ Read Audio | Receive Responses ].
+	for !exitFlag {
+		// Exit from loop if exitFlag condition is met.
+
+		// Wait for Goroutines to finish before going to next loop.
+		var wg sync.WaitGroup
+		wg.Add(1)
+
+		// Start stream.
+		if streamIsNull {
```

Comment:
`streamIsNull` is always true here. So, I don't think we need the `if`.

Reply:
Do you really read the code?   At the beginning it will be true to initialize the stream and when the program proceeds into the subsequent for loop it will be false to skip initialization of the stream.

Reply:
Yes. Sorry, it's very difficult for me to follow.